tests/utils/validate_jsonld_context.py

```diff
@@ -0,0 +1,117 @@
+import json
+import os
+
+from rdflib import Graph, term
+
+
+class RdfExpectations:
+    """Calculates the expected RDF graph.
+    It does not use for URI generation any of the code available in the `linkml`
+    package on purpose to avoid bug contagion!"""
+
+    _schema_yaml: dict = {}
+    _g: Graph = Graph()
+    _schema_uri: str = ""
+
+    def __init__(self, schema_yaml, schema_jsonld):
+        self._schema_yaml = schema_yaml
+        self._g = Graph().parse(data=schema_jsonld, format="json-ld")
+        if os.environ.get("DETAILED_OUTPUT"):
+            print("ℹ️ℹ️ℹ️ℹ️ℹ️ℹ️ℹ️ℹ️ℹ️ℹ️ℹ️ℹ️ℹ️ℹ️")
+            print("ℹ️ℹ️      JSON-LD       ℹ️ℹ️")
+            print(json.dumps(schema_jsonld, indent=2))
+            print("ℹ️ℹ️ℹ️ℹ️ℹ️ℹ️ℹ️ℹ️ℹ️ℹ️ℹ️ℹ️ℹ️ℹ️")
+            print("🐢🐢🐢🐢🐢🐢🐢🐢🐢🐢🐢🐢🐢🐢")
+            print("🐢🐢 RDF Graph (Turtle) 🐢🐢")
+            print(self._g.serialize(format="ttl"))
+            print("🐢🐢🐢🐢🐢🐢🐢🐢🐢🐢🐢🐢🐢🐢")
+
+    def expected_id_uri(self):
+
+        if "name" in self._schema_yaml.keys():
+            if ":" in self._schema_yaml["name"]:
+                prefix = self._schema_yaml["name"].split(":")[0]
+                self._schema_uri = self._schema_yaml["prefixes"][prefix] + self._schema_yaml["name"].split(":")[1]
+            elif "default_prefix" in self._schema_yaml.keys():
+                self._schema_uri = (
+                    self._schema_yaml["prefixes"][self._schema_yaml["default_prefix"]] + self._schema_yaml["name"]
+                )
+            else:
+                self._schema_uri = self._schema_yaml["id"] + "/" + self._schema_yaml["name"]
+        else:
+            self._schema_uri = self._schema_yaml["id"]
+        for subject, predicate in self._g.subject_predicates(
+            object=term.URIRef("https://w3id.org/linkml/SchemaDefinition")
+        ):
+            assert str(subject) == self._schema_uri
+            assert str(predicate) == "http://www.w3.org/1999/02/22-rdf-syntax-ns#type"
+        print(f"URI SchemaDefinition: {self._schema_uri}")
```

Comment:
You are already testing `self._schema_uri` with an assertion -- no need to clutter up the test output with a `print` statement.

Reply:
The pytest error message if the assertion fails should be enough for diagnosing errors, or at least for knowing where to set breakpoints for the debugger.

Reply:
I prefer printouts over debuggers, but I have no horse in this race either so I'll simply remove it.

Reply:
Done